libraries/SPI/SPI.cpp

```diff
@@ -74,13 +75,17 @@ void SPIClass::setHwCs(bool use) {
 }
 
 void SPIClass::beginTransaction(SPISettings settings) {
-    while(SPI1CMD & SPIBUSY) {}
+    while(inTransaction || SPI1CMD & SPIBUSY) {
```

Comment:
I don't think this is correct. 
`inTransaction` may only be set to `false` by a call to `endTransaction`. But how will `endTransaction` be called if you are spinning inside this loop?
Could you please share a simple sketch which illustrates the issue fixed by this change?